server/service/schedule/schedule.go

```diff
@@ -517,8 +550,30 @@ func (s *Schedule) holdLock() (bool, context.CancelFunc) {
 	return true, cancelFn
 }
 
-func (s *Schedule) getLatestStats() (fleet.CronStats, error) {
-	return s.statsStore.GetLatestCronStats(s.ctx, s.name)
+func (s *Schedule) getLatestStats() (fleet.CronStats, fleet.CronStats, error) {
+	var scheduled, triggered fleet.CronStats
+
+	cs, err := s.statsStore.GetLatestCronStats(s.ctx, s.name)
+	if err != nil {
+		return fleet.CronStats{}, fleet.CronStats{}, err
+	}
+	if len(cs) > 2 {
+		return fleet.CronStats{}, fleet.CronStats{}, fmt.Errorf("get latest stats expected length to be no more than two but got length: %d", len(cs))
+	}
+
+	for _, stats := range cs {
+		switch stats.StatsType {
+		case fleet.CronStatsTypeScheduled:
+			scheduled = stats
+		case fleet.CronStatsTypeTriggered:
+			triggered = stats
+		default:
+			// this should not happen so log for debugging
+			level.Debug(s.logger).Log("msg", fmt.Sprintf("get latest stats unexpected type: %s", stats.StatsType))
```

Comment:
This should probably be an error, as it is probably related to a bug. Debug logs will most likely be ignored...